File: backend/src/domain/dto/experience.py
from typing import Optional

from pydantic import BaseModel


class Experience(BaseModel):
    """Professional experience with position, company and duration."""

    position: str
    company: str
    location: str
    website: Optional[str] = None
    logo: Optional[str] = None
    description: str
    skills: str
    duration: str

    def to_dict(self):
        """Return a dictionary representation of this experience."""
        return {
            "position": self.position,
            "company": self.company,
            "location": self.location,
            "website": self.website,
            "logo": self.logo,
            "description": self.description,
            "skills": self.skills,
            "duration": self.duration,
        }

File: frontend/src/interfaces/Experience.ts
export interface Experience {
    position: string;
    company: string;
    period?: string;
    location: string;
    website?: string;
    logo?: string;
    actual_job?: boolean;
    skills: string;
    description: string;
    start_date?: string; 
    duration?: string;
  }

// Interface para os dados de duração por empresa
export interface CompanyDuration {
    name: string;
    duration: string;
  }
  
